halibot/message.py

```diff
@@ -26,12 +26,26 @@ def makeOrigin(self):
 
 class Message():
 
+	class Author(str):
+		def __init__(self, a, id=""):
+			self.id = id
+			self.name = str(a) if a else ""
+		def __str__(self):
+			return self.name
+		def __repr__(self):
+			return self.name
+		def __setattr__(self, key, value):
+			if key in ("id", "name"):
+				object.__setattr__(self, key, value)
+		def __getattr__(self, key):
+			return str(object.__getattr__(self,key))
+
 	def __init__(self, **kwargs):
 		self.log = logging.getLogger(self.__class__.__name__)
 
 		self.body = kwargs.get('body', None)
 		self.type = kwargs.get('type', 'simple')
-		self.author = kwargs.get('author', None)
+		self.author = self.Author(kwargs.get('author', None))
```

Comment:
If we are casting the name to a string, maybe we should not have `None` be the default?

Note that `str(None) == "None"`